File: bloock-bridge/src/entity_mappings/network.rs
use crate::items::Network;
use bloock_core::config::entity::network::Network as CoreNetwork;

impl From<Network> for CoreNetwork {
    fn from(n: Network) -> Self {
        match n {
            Network::BloockChain => CoreNetwork::BloockChain,
            Network::EthereumMainnet => CoreNetwork::EthereumMainnet,
            Network::EthereumGoerli => CoreNetwork::EthereumGoerli,
            Network::GnosisChain => CoreNetwork::GnosisChain,
        }
    }
}

pub fn map_network_from_i32(network: i32) -> Option<CoreNetwork> {
    Network::from_i32(network).map(|network| network.into())
}

File: languages/java/lib/src/main/java/bloock/sdk/java/entity/Network.java
package bloock.sdk.java.entity;

import bloock.sdk.java.bridge.proto.Config;

public enum Network {
  ETHEREUM_MAINNET,
  ETHEREUM_RINKEBY,
  ETHEREUM_GOERLI,
  GNOSIS_CHAIN,
  BLOOCK_CHAIN;

  public Config.Network toProto() {
    switch (this) {
      case BLOOCK_CHAIN:
        return Config.Network.BLOOCK_CHAIN;
      case ETHEREUM_GOERLI:
        return Config.Network.ETHEREUM_GOERLI;
      case ETHEREUM_MAINNET:
        return Config.Network.ETHEREUM_MAINNET;
      case ETHEREUM_RINKEBY:
        return Config.Network.ETHEREUM_RINKEBY;
      case GNOSIS_CHAIN:
        return Config.Network.GNOSIS_CHAIN;
      default:
        return Config.Network.BLOOCK_CHAIN;
    }
  }
}
